frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseGitMacheteRepositoryReadyAction.java

```diff
@@ -7,7 +7,7 @@
 import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeyGitMacheteRepository;
 import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
 
-public abstract class BaseGitMacheteRepositoryReadyAction extends BaseProjectKeyAvailabilityAssuranceAction
+public abstract class BaseGitMacheteRepositoryReadyAction extends BaseProjectDependentAction
```

Comment:
Basically, there's *no access to the project key* (`getProject`) unless we're first sure that the project key is actually available - and this check only ever happens in `BaseProjectDependentAction#update`.

Most of the time the project is available... this is only covering corner cases around opening the project.